lte/gateway/c/oai/lib/mobility_client/MobilityClientAPI.cpp

```diff
@@ -69,79 +70,61 @@ int get_assigned_ipv4_block(
 
 int pgw_handle_allocate_ipv4_address(
     const char* subscriber_id, const char* apn, struct in_addr* addr,
-    itti_sgi_create_end_point_response_t sgi_create_endpoint_resp,
-    const char* pdn_type, spgw_state_t* spgw_state,
-    s_plus_p_gw_eps_bearer_context_information_t* new_bearer_ctxt_info_p,
-    s5_create_session_response_t s5_response) {
+    const char* pdn_type, teid_t context_teid, ebi_t eps_bearer_id) {
   MobilityServiceClient::getInstance().AllocateIPv4AddressAsync(
       subscriber_id, apn,
-      [=, &s5_response](
-          const Status& status, AllocateIPAddressResponse ip_msg) {
+      [addr, subscriber_id, apn, pdn_type, context_teid, eps_bearer_id](
```

Comment:
We do not need to capture the addr from the caller scope here. It was originally used to capture the addr allocated by the RPC, but this will be taken care of through the ITTI message now.